pkg/cloudprovider/events/events.go

```diff
@@ -0,0 +1,40 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package events
+
+import (
+	v1 "k8s.io/api/core/v1"
+
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/events"
+)
+
+func ProvisionerFailedToResolveNodeTemplate(provisioner *v1alpha5.Provisioner) events.Event {
+	return events.Event{
+		InvolvedObject: provisioner,
+		Type:           v1.EventTypeWarning,
+		Message:        "Failed to resolve AWSNodeTemplate",
+		DedupeValues:   []string{provisioner.Name},
+	}
+}
+
+func MachineFailedToResolveNodeTemplate(machine *v1alpha5.Machine) events.Event {
+	return events.Event{
+		InvolvedObject: machine,
+		Type:           v1.EventTypeWarning,
```

Comment:
Is this a Warning? I think this should probably be an `Error` event?

Reply:
There are only Warning and Normal events, right? https://pkg.go.dev/k8s.io/api@v0.26.6/core/v1#:~:text=const%20(%0A%09//%20Information%20only%20and%20will%20not%20cause%20any%20problems%0A%09EventTypeNormal%20string%20%3D%20%22Normal%22%0A%09//%20These%20events%20are%20to%20warn%20that%20something%20might%20go%20wrong%0A%09EventTypeWarning%20string%20%3D%20%22Warning%22%0A)